src/widgets/modals/ByProgramModal.js

```diff
@@ -31,11 +36,15 @@ import { FlexColumn } from '../FlexColumn';
 
 const { THIS_STORE_TAGS, THIS_STORE_NAME_ID } = SETTINGS_KEYS;
 
-const modalProps = ({ dispatch, program, orderType }) => ({
+const modalProps = ({ dispatch, program, orderType, customer }) => ({
   program: {
     queryStringSecondary: 'name CONTAINS[c] $0',
     onSelect: value => dispatch(selectProgram(value)),
   },
+  customer: {
+    secondaryFilterProperty: 'code',
+    onSelect: value => dispatch(selectCustomer(value)),
+  },
```

Comment:
this `modalProps` function passes the props for the `AutocompleteSelector` modal, for choosing out of the options for the current step. The primary filter property default is to filter by `'name'`, and the `onSelect` being passed will just trigger the reducer, passing back the `customer` record and storing it in state, as well as resetting all other steps.

For the actual data filling the modal for selecting a customer, it is below in a function `getCustomers`